tasktiger/redis_scripts.py

```diff
@@ -469,20 +480,39 @@ def execute_pipeline(self, pipeline, client=None):
         results = redis_scripts.execute_pipeline(p)
         """
 
+        client = client or self.redis
+
+        executing_pipeline = None
         try:
+
             # Prepare args
             stack = pipeline.command_stack
-            script_args = [len(stack)]
+            script_args = [int(self.can_replicate_commands), len(stack)]
             for args, options in stack:
                 script_args += [len(args)-1] + list(args)
 
-            # Make sure scripts exist
-            if pipeline.scripts:
-                pipeline.load_scripts()
-
             # Run the pipeline
-            raw_results = self._execute_pipeline(args=script_args,
-                                                 client=client)
+            if self.can_replicate_commands:  # Redis 3.2 or higher
```

Comment:
Are we sure after `redis.replicate_commands()` is called a `_G['f_' .. sha]()` call is turned back into a regular `EVALSHA` command when it is prepared to be sent over to the slave node?  Just want to make sure it isn't silently ignored.

Reply:
Yes.

```
In [1]: s1 = app.redis.register_script('''redis.call('incr', 'x')''')

In [2]: s2 = app.redis.register_script('''redis.replicate_commands(); _G['f_{}']()'''.format(s1.sha))

In [3]: app.redis.script_load(s1.script)
Out[3]: '055c90c3af205164e578f6e1cb7ee167b810c6dd'

In [4]: app.redis.script_load(s2.script)
Out[4]: 'b7b28fdcb955f97bea62c60cc0f83be0f8ae6704'

# Restart slave here to empty its scripts cache and run MONITOR

In [5]: s2()

# Slave output:

MONITOR
+OK
+1515093425.210639 [0 127.0.0.1:6379] "SELECT" "0"
+1515093425.210681 [0 127.0.0.1:6379] "MULTI"
+1515093425.210689 [0 127.0.0.1:6379] "incr" "x"
+1515093425.210702 [0 127.0.0.1:6379] "EXEC"
```